daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/VariableMap1.scala

```diff
@@ -324,17 +348,31 @@ class VariableMap private(vTable: Map[GlobalQName, ArrayBuffer[VariableInstance]
   // have default value expressions or are defined externally.
   def forceExpressionEvaluations(state: ParseOrUnparseState): Unit = {
     vTable.foreach { case (_, variableInstances) => { variableInstances.foreach { inst => {
-      (inst.state, inst.value, inst.defaultValueExpr.isDefined) match {
+      (inst.state, inst.defaultValueExpr.isDefined) match {
         // Evaluate defineVariable statements with non-constant default value expressions
-        case (VariableDefined, DataValue.NoValue, true) => {
+        case (VariableUndefined, true) => {
           val res = inst.defaultValueExpr.get.evaluate(state)
-          inst.setValue(DataValue.unsafeFromAnyRef(res))
+          inst.setDefaultValue(DataValue.unsafeFromAnyRef(res))
         }
-        case (_, _, _) => // Do nothing
+        case (_, _) => // Do nothing
       }
     }}}}
   }
 
+
+  /**
+   * This function is called immediately after forceExpressionEvaluations in
+   * order to set the firstInstanceInitialValue for each variable instance.
+   * These initial values will be inherited by future new variable instances if
+   * the newVariableInstance statement does not provide a default value
+   * expression
+   */
+  def setFirstInstanceInitialValues(): Unit = {
+    vTable.foreach { case (_, variableInstances) => {
+      variableInstances.foreach { inst => inst.firstInstanceInitialValue = inst.value }
```

Comment:
variableInstances should always have a length of 1 here, right? Since this should only ever be called before NVI's are evaluated? Thoughts on making this
```scala
Assert.invariant(variableInstances.length == 1)
variablesInstances(0).firstInstanceInitialValue = inst._value
```
Note that if firstInstanceInitialValue is turned into an Option mentioend above, then this is just a straigt copy of the private _value. So it keeps the Some/None aspect of the value as well if a instance was never set by a default.

Reply:
Makes sense and I imagine that would be more performant anyway.